crates/bevy_math/src/bounding/bounded2d/primitive_impls.rs

```diff
@@ -183,6 +183,38 @@ impl Bounded2d for Ellipse {
     }
 }
 
+impl Bounded2d for Rhombus {
+    fn aabb_2d(&self, translation: Vec2, rotation: impl Into<Rotation2d>) -> Aabb2d {
+        let rotation_mat = rotation.into();
+
+        let [a, b, c, d] = [
+            rotation_mat * Vec2::new(self.half_diagonals.x, 0.0) + translation,
+            rotation_mat * Vec2::new(0.0, self.half_diagonals.y) + translation,
+            rotation_mat * Vec2::new(-self.half_diagonals.x, 0.0) + translation,
+            rotation_mat * Vec2::new(0.0, -self.half_diagonals.y) + translation,
+        ];
+
+        let min = Vec2::new(
+            a.x.min(b.x).min(c.x).min(d.x),
+            a.y.min(b.y).min(c.y).min(d.y),
+        );
+        let max = Vec2::new(
+            a.x.max(b.x).max(c.x).max(d.x),
+            a.y.max(b.y).max(c.y).max(d.y),
+        );
+
+        Aabb2d { min, max }
```

Comment:
You can factor out the translations out of the minima and maxima, like so:
```suggestion
        let [a, b, c, d] = [
            rotation_mat * Vec2::new(self.half_diagonals.x, 0.0),
            rotation_mat * Vec2::new(0.0, self.half_diagonals.y),
            rotation_mat * Vec2::new(-self.half_diagonals.x, 0.0),
            rotation_mat * Vec2::new(0.0, -self.half_diagonals.y),
        ];

        let min = Vec2::new(
            a.x.min(b.x).min(c.x).min(d.x),
            a.y.min(b.y).min(c.y).min(d.y),
        ) + translation;
        let max = Vec2::new(
            a.x.max(b.x).max(c.x).max(d.x),
            a.y.max(b.y).max(c.y).max(d.y),
        ) + translation;

        Aabb2d { min, max }
```
Then you can notice that `a = -c` and `b = -d` (as per the linearity of matrix multiplication and the fact that the input vectors are just sign, flipped), meaning we can skip half the matrix multiplications.
```suggestion
        let [a, b] = [
            rotation_mat * Vec2::new(self.half_diagonals.x, 0.0),
            rotation_mat * Vec2::new(0.0, self.half_diagonals.y),
        ];

        let min = Vec2::new(
            a.x.min(b.x).min(-a.x).min(-b.x),
            a.y.min(b.y).min(-a.y).min(-b.y),
        ) + translation;
        let max = Vec2::new(
            a.x.max(b.x).max(-a.x).max(-b.x),
            a.y.max(b.y).max(-a.y).max(-b.y),
        ) + translation;

        Aabb2d { min, max }
```
Taking the maximum between `a` and `-a` amounts to taking the absolute value.
Taking the minimum between `a` and `-a` amounts to taking minus the absolute value.
```suggestion
        let [a, b] = [
            rotation_mat * Vec2::new(self.half_diagonals.x, 0.0),
            rotation_mat * Vec2::new(0.0, self.half_diagonals.y),
        ];

        let min = Vec2::new(
            (-a.x.abs()).min(-b.x.abs()),
            (-a.y.abs()).min(-b.y.abs()),
        ) + translation;
        let max = Vec2::new(
            (a.x.abs()).max(b.x.abs()),
            (a.y.abs()).max(b.y.abs()),
        ) + translation;

        Aabb2d { min, max }
```
Factoring out the minus signs by realising `min(-x, -y) = -max(x,y)`, we find out that min and max are basically the same up to a sign.
We can also simplify the operations by using the elementwise operations on Vec2's.
```suggestion
        let [rotated_x_half_diagonal, rotated_y_half_diagonal] = [
            rotation_mat * Vec2::new(self.half_diagonals.x, 0.0),
            rotation_mat * Vec2::new(0.0, self.half_diagonals.y),
        ];

        let aabb_half_extent = rotated_x_half_diagonal.abs().max(rotated_y_half_diagonal.abs());

        Aabb2d { -aabb_half_extent + translation , aabb_half_extent + translation }
```
IIRC there was an even cleverer trick, that simply took the absolute value of the matrix [I think it's used in rectangle code].
But I'm not so sure about its derivation right now.